docs/types/traits-and-interfaces.md

```diff
@@ -81,6 +123,92 @@ interface HasName
 
 Here, `HasName` looks a lot like `Named`, except it's an interface instead of a trait. This means both `Bob` and `Larry` provide `HasName`! The programmers that wrote `Bob` and `Larry` don't even have to be aware that `HasName` exists.
 
-Pony interfaces can have functions with default implementations as well. A type will only pick those up if it explicitly declares that it `is` that interface.
+## Differences between traits and interfaces
+
+It is common for new Pony users to ask, __Should I use traits or interfaces in my own code?__ Both! Interfaces are more flexible, so if you're not sure what you want, use an interface. But traits are a powerful tool as well.
+
+### Private methods
+
+A key difference between traits and interfaces is that interfaces can't have private methods. So, if you need private methods, you'll need to use a trait and have users opt in via nominal typing. Interfaces can't have private methods because otherwise, users could use them to break encapsulation and access private methods on concrete types from other packages. For example:
+
+```pony
+actor Main
+  new create(env: Env) =>
+    let x: String ref = "sailor".string()
+    let y: Foo = x
+    y._set(0, 'f')
+    env.out.print("Hello, " + x)
+
+interface Foo
+  fun ref _set(i: USize, value: U8): U8
+```
+
+In the code above, the interface `Foo` allows access to the private `_set` method of String and allows for changing `sailor` to `failor` or it would anyway, if interfaces were allowed to have private methods.
```

Comment:
```suggestion
In the code above, the interface `Foo` allows access to the private `_set` method of `String` and allows for changing `sailor` to `failor` or it would anyway, if interfaces were allowed to have private methods.
```